lib/utils/generateExtractFieldsFn.js

```diff
@@ -0,0 +1,22 @@
+'use strict';
+
+const R = require('ramda');
+
+module.exports = function generateExtractFieldsFn (mappings) {
```

Comment:
Seems too small to keep in its own file. Whats more, it's related enough to the `matchScriptData` function that they could live both on the same module, and we should come up with a good name for that module.

Reply:
Any suggestions?

Reply:
Maybe `scriptData` for the module name, so the code would look like

```js
request(options, opts.throttle)
    .then(scriptData.parse)
    .then(scriptData.extractor(MAPPINGS))
    // ...
```

(note I use `extractor` rather than `extract` because that function returns the actual function that does the extraction)

Reply:
> Similar method passes the test :)

no similar method doesn't work yet!